### ISSUE TITLE
Old messages keep being reprocessed

### ISSUE THREAD

**Issue by lukas-ostrovskis** [rsverhagen94/TUD-Collaborative-AI-2023#6]
RescueBot keeps processing messages that it already acted upon. This results in unexpected behavior in various places in the code (which I assume has been noticed previously as there are multiple places with the comment "bug fix" where the received messages are cleared).

A simple example of this:
![image](https://user-images.githubusercontent.com/63771430/223108863-d57aa784-595b-4983-a62b-0f2c414a08c2.png)

When a human agent sends a message that a mildly injured victim was found, RescueBot keeps processing that message every frame and fills up the todo list with that same victim. Of course, this specific example can be solved by simply checking if the victim isn't already in the todo list before appending it.

Question is: why do already processed messages keep getting reprocessed, is this necessary anywhere? Would it be possible to update messages with some kind of "processed" status and ignore them during each subsequent call of `_processMessages`?

**Comment by rsverhagen94** [comment]
I don't believe it is necessary that already processed messages keep getting reprocessed. If you want to remove this or test whether reprocessing has a function somewhere, you could add a few lines to ```_processMessages``` to only process messages not processed before.
